src/main/drivers/compass/compass.h

```diff
@@ -36,4 +36,5 @@ typedef struct magDev_s {
     fp_rotationMatrix_t rotationMatrix;
     ioTag_t magIntExtiTag;
     int16_t magGain[3];
+    uint16_t magOdrHz;
```

Comment:
I have thought about user-configurable 'intended ODR' CLI value, but rejected it as being unnecessary, and out of the scope of this PR.

It would be complex to implement because we would have to get the user's intended ODR and compare that to the capabilities of the Mag being used, then select the highest ODR supported by the Mag that was equal to or less than the user's intended ODR.

In practice, we want the fastest ODR possible, especially in noisy environments, because the higher the ODR, the less likely it is that an extreme transient can cast a prolonged effect on the reported value, and the more likely it is to get a good calibration result.

So the current code simply gets the highest ODR supported by the hardware, which is 200Hz for the QMC units, and quite a bit less for other Mags.

What the `magOdrHz` value provides is that the active ODR value in the driver is used to set the task rate.  Even if the user could set their intended ODR, we would still need to retain this `magOdrHz` value, since the value that the hardware will run at may not be the value the user may have intended.  For example, the user may request 200Hz but the Mag may only support 50Hz, in which case we would need `magOdrHz` to be 50, not 200, so that the task gets optimised to 50hz.  Hence, in addition to the existing`magOdrHz`, we would require a separate value for 'requested ODR' in the CLI.

Hence I don't think it would be useful for the user to set their requested ODR.

Future PRs that utilise mag data may want to know the data ODR rate in order to optimally smooth Mag data, and the `magOdrHz` value may be useful in those PR's also.